### ISSUE TITLE
Unable to create new posts

### ISSUE THREAD

**Issue by SixBytesUnder** [TryGhost/Ghost#11231]
### Issue Summary

Upgraded to 2.35.0 today and wanted to add a new post. As soon as Ghost tries to save draft of my new post, I get a red bar at the top with error message "An error occurred Unable to update nested relation. [object Object]"
Browser console shows a request 
https://mydomain.com/ghost/api/canary/admin/posts/ 500
from file
vendor.min-8a9c6f8a3daf9a5470d26dae32d1c192.js:850

No new post is created.
Also tried in incognito mode, to make sure no old cookies are present.

### To Reproduce

1. Log in to Ghost admin
2. Click: Manage -> Posts -> New post
3. Enter any title and focus out or start writing your post
4. As soon as Ghost tries to save a draft, error occurs. New post does not get created.

### Technical details:

* Ghost Version: 2.35.0 (self hosted)
* Node Version: v10.16.0
* Browser/OS: Version 77.0.3865.90 (Official Build) (64-bit) / Windows 10
* Database: MariaDB 10.4.8
* System: CentOS 7

Update: technical details

**Comment by kevinansfield** [comment]
@SixBytesUnder can you please share the server-side log for the failed request?

**Comment by SixBytesUnder** [comment]
@kevinansfield here you go.
Should I hide domain and IP addresses?

```
{ 
   "name":"Log",
   "hostname":"ns304241.ip-94-23-210.eu",
   "pid":16729,
   "level":50,
   "req":{ 
      "meta":{ 
         "requestId":"6b5ba357-e7cb-4a7a-9b0e-bd4ea83bfdcc",
         "userId":"1"
      },
      "url":"/posts/",
      "method":"POST",
      "originalUrl":"/ghost/api/canary/admin/posts/",
      "params":{ 

      },
      "headers":{ 
         "x-real-ip":"162.158.159.60",
         "host":"allurcode.com",
         "connection":"close",
         "content-length":"1301",
         "accept-encoding":"gzip",
         "cf-ipcountry":"GB",
         "x-forwarded-for":"146.198.159.15",
         "cf-ray":"5240ee44ebe9e694-LHR",
         "x-forwarded-proto":"https",
         "cf-visitor":"{\"scheme\":\"https\"}",
         "sec-fetch-mode":"cors",
         "origin":"https://allurcode.com",
         "app-pragma":"no-cache",
         "user-agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36",
         "content-type":"application/json; charset=UTF-8",
         "accept":"application/json, text/javascript, */*; q=0.01",
         "x-requested-with":"XMLHttpRequest",
         "x-ghost-version":"2.35",
         "sec-fetch-site":"same-origin",
         "referer":"https://allurcode.com/ghost/",
         "accept-language":"en-GB,en;q=0.9,pl;q=0.8",
         "cookie":"**REDACTED**",
         "cf-connecting-ip":"146.198.159.15",
         "cdn-loop":"cloudflare"
      },
      "query":{ 

      }
   },
   "res":{ 
      "_headers":{ 
         "x-powered-by":"Express",
         "cache-control":"no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
         "access-control-allow-origin":"https://allurcode.com",
         "vary":"Origin, Accept-Encoding",
         "content-type":"application/json; charset=utf-8",
         "content-length":"251",
         "etag":"W/\"fb-4QrI0+lE96l1uJEoXLOp4T0QJo8\""
      },
      "statusCode":500,
      "responseTime":"79ms"
   },
   "err":{ 
      "id":"fd2048c0-ec24-11e9-964b-f974cfd8b91a",
      "domain":"https://allurcode.com",
      "code":"UPDATE_RELATION",
      "name":"BookshelfRelationsError",
      "statusCode":500,
      "level":"critical",
      "message":"Unable to update nested relation.",
      "context":"{\"key\":\"authors\",\"tableName\":\"posts\",\"method\":\"setBelongsToMany\"}",
      "stack":"BookshelfRelationsError: Unable to update nested relation.\n    at new BookshelfRelationsError (/var/www/allurcode.com/versions/2.35.0/node_modules/bookshelf-relations/errors.js:10:26)\n    at setter.bind.catch (/var/www/allurcode.com/versions/2.35.0/node_modules/bookshelf-relations/lib/relations.js:70:27)\n    at tryCatcher (/var/www/allurcode.com/versions/2.35.0/node_modules/bookshelf-relations/node_modules/bluebird/js/release/util.js:16:23)\n    at Promise._settlePromiseFromHandler (/var/www/allurcode.com/versions/2.35.0/node_modules/bookshelf-relations/node_modules/bluebird/js/release/promise.js:517:31)\n    at Promise._settlePromise (/var/www/allurcode.com/versions/2.35.0/node_modules/bookshelf-relations/node_modules/bluebird/js/release/promise.js:574:18)\n
   at Promise._settlePromise0 (/var/www/allurcode.com/versions/2.35.0/node_modules/bookshelf-relations/node_modules/bluebird/js/release/promise.js:619:10)\n    at Promise._settlePromises (/var/www/allurcode.com/versions/2.35.0/node_modules/bookshelf-relations/node_modules/bluebird/js/release/promise.js:695:18)\n    at _drainQueueStep (/var/www/allurcode.com/versions/2.35.0/node_modules/bookshelf-relations/node_modules/bluebird/js/release/async.js:138:12)\n    at _drainQueue (/var/www/allurcode.com/versions/2.35.0/node_modules/bookshelf-relations/node_modules/bluebird/js/release/async.js:131:9)\n
   at Async._drainQueues (/var/www/allurcode.com/versions/2.35.0/node_modules/bookshelf-relations/node_modules/bluebird/js/release/async.js:147:5)\n    at Immediate.Async.drainQueues [as _onImmediate] (/var/www/allurcode.com/versions/2.35.0/node_modules/bookshelf-relations/node_modules/bluebird/js/release/async.js:17:14)\n    at runCallback (timers.js:705:18)\n    at tryOnImmediate (timers.js:676:5)\n    at processImmediate (timers.js:658:5)\n    at process.topLevelDomainCallback (domain.js:126:23)\n\nError: ER_NO_REFERENCED_ROW_2: Cannot add or update a child row: a foreign key constraint fails (`allurcode_com`.`posts_authors`, CONSTRAINT `posts_authors_post_id_foreign` FOREIGN KEY (`post_id`) REFERENCES `posts` (`id`))\n    at Query.Sequence._packetToError (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)\n    at Query.ErrorPacket (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/sequences/Query.js:77:18)\n    at Protocol._parsePacket (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/Protocol.js:291:23)\n    at Parser._parsePacket (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/Parser.js:433:10)\n    at Parser.write (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/Parser.js:43:10)\n    at Protocol.write (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/Protocol.js:38:16)\n    at Socket.<anonymous> (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/Connection.js:91:28)\n    at Socket.<anonymous> (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/Connection.js:525:10)\n    at Socket.emit (events.js:198:13)\n    at Socket.EventEmitter.emit (domain.js:448:20)\n    at addChunk (_stream_readable.js:288:12)\n    at readableAddChunk (_stream_readable.js:269:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)\n
  --------------------\n    at Protocol._enqueue (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/Protocol.js:144:48)\n    at Connection.query (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/Connection.js:201:25)\n    at /var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/dialects/mysql/index.js:130:18\n    at Promise._execute (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/debuggability.js:427:9)\n    at Promise._resolveFromExecutor (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:518:18)\n
    at new Promise (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:103:10)\n    at Client_MySQL._query (/var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/dialects/mysql/index.js:124:12)\n    at Client_MySQL.query (/var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/client.js:158:17)\n    at /var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/transaction.js:331:24\n    at Promise._execute (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/debuggability.js:427:9)\n    at Promise._resolveFromExecutor (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:518:18)\n    at new Promise (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:103:10)\n    at Client_MySQL.trxClient.query (/var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/transaction.js:326:12)\n    at Runner.query (/var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/runner.js:135:36)\n    at /var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/runner.js:39:23\n    at tryCatcher (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/util.js:16:23)\n    at /var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/using.js:185:26\n    at tryCatcher (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/util.js:16:23)\n    at Promise._settlePromiseFromHandler (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:547:31)\n    at Promise._settlePromise (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:604:18)\n    at Promise._settlePromise0 (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:649:10)\n    at Promise._settlePromises (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:729:18)\n    at Promise._fulfill (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:673:18)\n    at PromiseArray._resolve (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise_array.js:127:19)\n    at PromiseArray._promiseFulfilled (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise_array.js:145:14)\n    at Promise._settlePromise (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:609:26)\n    at Promise._settlePromise0 (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:649:10)\n    at Promise._settlePromises (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:729:18)"
   },
   "msg":"Unable to update nested relation.",
   "time":"2019-10-11T12:45:22.154Z",
   "v":0
}
```

**Comment by SixBytesUnder** [comment]
Also last error after running `ghost log -n 50 -e` is
```
Error: ER_NO_REFERENCED_ROW_2: Cannot add or update a child row: a foreign key constraint fails (`allurcode_com`.`posts_authors`, CONSTRAINT `posts_authors_post_id_foreign` FOREIGN KEY (`post_id`) REFERENCES `posts` (`id`))
    at Query.Sequence._packetToError (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Query.ErrorPacket (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/sequences/Query.js:77:18)
    at Protocol._parsePacket (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/Parser.js:43:10)
    at Protocol.write (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/Connection.js:91:28)
    at Socket.<anonymous> (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/Connection.js:525:10)
    at Socket.emit (events.js:198:13)
    at Socket.EventEmitter.emit (domain.js:448:20)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
    --------------------
    at Protocol._enqueue (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/protocol/Protocol.js:144:48)
    at Connection.query (/var/www/allurcode.com/versions/2.35.0/node_modules/mysql/lib/Connection.js:201:25)
    at /var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/dialects/mysql/index.js:130:18
    at Promise._execute (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/debuggability.js:427:9)
    at Promise._resolveFromExecutor (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:518:18)
    at new Promise (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:103:10)
    at Client_MySQL._query (/var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/dialects/mysql/index.js:124:12)
    at Client_MySQL.query (/var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/client.js:158:17)
    at /var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/transaction.js:331:24
    at Promise._execute (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/debuggability.js:427:9)
    at Promise._resolveFromExecutor (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:518:18)
    at new Promise (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:103:10)
    at Client_MySQL.trxClient.query (/var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/transaction.js:326:12)
    at Runner.query (/var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/runner.js:135:36)
    at /var/www/allurcode.com/versions/2.35.0/node_modules/knex/lib/runner.js:39:23
    at tryCatcher (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/util.js:16:23)
    at /var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/using.js:185:26
    at tryCatcher (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:547:31)
    at Promise._settlePromise (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:729:18)
    at Promise._fulfill (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:673:18)
    at PromiseArray._resolve (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise_array.js:127:19)
    at PromiseArray._promiseFulfilled (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise_array.js:145:14)
    at Promise._settlePromise (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:609:26)
    at Promise._settlePromise0 (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/var/www/allurcode.com/versions/2.35.0/node_modules/bluebird/js/release/promise.js:729:18)
```

**Comment by SixBytesUnder** [comment]
@kevinansfield if it makes any difference I was upgrading from 2.30.2 to 2.35.0 and it looks like maybe some database migration didn't run on my update. What can I do?
FYI, updating existing posts (drafts) doesn't save changes, but fails silently, without any error message.

**Comment by Szandor72** [comment]
I've run into the same issue, I could write and edit posts for a while and then it suddenly stopped with the same error.

**Comment by SixBytesUnder** [comment]
Hi @gargol 
Running `select * from migrations where version like '2.3%';` gives me

id | name | version | currentVersion
-- | -- | -- | --
29 | 1-add-webhook-columns.js | 2.3 | 2.6
30 | 2-add-webhook-edit-permission.js | 2.3 | 2.6
58 | 1-remove-name-and-password-from-members-table.js | 2.31 | 2.35
59 | 01-add-members-stripe-customers-table.js | 2.32 | 2.35
60 | 02-add-name-to-members-table.js | 2.32 | 2.35
61 | 01-correct-members-stripe-customers-table.js | 2.33 | 2.35
62 | 01-add-stripe-customers-subscriptions-table.js | 2.34 | 2.35
63 | 02-add-email-to-members-stripe-customers-table.js | 2.34 | 2.35
64 | 03-add-name-to-members-stripe-customers-table.js | 2.34 | 2.35
65 | 01-add-note-to-members-table.js | 2.35 | 2.35

Regarding logs, can you give me a hint how to find the logs you need? Are they in a particular file? Is there a parameter for `ghost log` that will give the results you're looking for?

**Comment by gargol** [comment]
@SixBytesUnder you might be looking in the logs located in `you_ghost_dir/content/log` and usually migration starts with a backup, so can look for `Creating database backup` or one of migration specific messages `Adding table: members_stripe_customers_subscriptions`. 

Another interesting bit to inspect would be the data that's causing the foreign key constraint failure for `posts_authors_post_id_foreign` key. Could you start up ghost instance with these debug variables `DEBUG=knex:(query|bindings)*` and check what data is being inserted when you save the post (the information should also be in the log).

I wasn't able to reproduce the same behavior so far when migrating from 2.30.2 to 2.35.0

**Comment by SixBytesUnder** [comment]
The issue fixed itself after updating Ghost to version 3.1.1
Thanks for looking into this.
